ee/clickhouse/queries/person_query.py

```diff
@@ -109,12 +118,54 @@ def _get_person_filters(self) -> Tuple[str, Dict]:
             expr, prop_params = prop_filter_json_extract(
                 property,
                 index,
-                prepend="personquery",
+                prepend=prepend,
                 allow_denormalized_props=True,
                 transform_expression=lambda column_name: f"argMax(person.{column_name}, _timestamp)",
+                property_operator=operator,
             )
 
             conditions.append(expr)
             params.update(prop_params)
 
-        return " ".join(conditions), params
+        if conditions:
+            # remove the first operator
+            return " ".join(conditions).replace(operator, "", 1), params
+
+        return "", params
+
+    def parse_grouped_properties(
```

Comment:
similar to `parse_prop_group_clauses`, but different enough that I thought combining them together would be a huuge PITA.

Reply:
Not to mention more confusing, since more options on that single function = harder to parse whats happening

Reply:
I don't get the intent here though - we're now creating separate subsystems for query building in queries? This again just makes the system worse.

Weren't the original problems solved by the EXCLUDE mode option passed.

Reply:
The way it was working was: EXCLUDE excluded person props from whichever query, but that didn't help _here_, where we want to get only the person properties.

I like the idea of a new logic component figuring this out, and can possible get rid of this TODO as well, maybe:

https://github.com/PostHog/posthog/blob/master/ee/clickhouse/models/property.py#L176

This exists^ because we don't have this logic module to decide how to generate the person query. Am I understanding this correctly?

Reply:
> This exists^ because we don't have this logic module to decide how to generate the person query. Am I understanding this correctly?

No it exists due to code we haven't refactored in other ways before

> The way it was working was: EXCLUDE excluded person props from whichever query, but that didn't help here, where we want to get only the person properties.

It is an essential part - exclude excluded the person props in the outer events query layer and column_optimizer told this query what person properties were used and needed to be queried.

---

Fundamentally you need to consider here is that you want to:
1. Minimize the amount of columns read
2. Minimize the amount of data being transferred between threads/do the filtering as deep as possible.

Both are needed to make this work in a performant way.

You probably need some sort of business logic layer here that:
1. Figures out what columns should be "forwarded" in the `PersonQuery` to outer layers
2. Figures out what columns can be filtered on in the `PersonQuery` (e.g. groups which are all person-property related).

Also don't forget to add a `performance` label to this PR as this can change benchmark results.